### PR TITLE
feat(frontend): avoid sorting if no balances store

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

It does not make sense to run the sorting by balance if the balance store is not initiated.

### CODE REVIEW

File: src/frontend/src/lib/derived/network-tokens.derived.ts [hunk 1]
```diff
@@ -25,13 +26,16 @@ export const combinedDerivedSortedNetworkTokens: Readable<Token[]> = derived(
...
-			$balances,
-			$exchanges
-		})
+		nonNullish($balances)
+			? pinTokensWithBalanceAtTop({
+					$tokens: $enabledNetworkTokens,
+					$balances,
+					$exchanges
+				})
+			: ($enabledNetworkTokens as TokenUi[])
```
Comment (peterpeterparker) [inline src/frontend/src/lib/derived/network-tokens.derived.ts:40]: Why this cast? Shouldn't we map the tokens? What if we add fields in TokenUI in the future?

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/derived/network-tokens.derived.ts]: Right! Moved internally using the `mapTokenUi` functions

**Comment by peterpeterparker** [comment]
It's a feature not refactoring. Updated the title.

**Review by peterpeterparker — COMMENTED** [review]
LGTM, thanks

Note that I assume you added the `.map` for the best performances, that's cool. To some extension, we could also return the result of the `.reduce()` which maps as well.

```
if (isNullish($balance)) {
  return [...positiveBalances, ...nonPositiveBalances]
}
```

**Comment by AntonioVentilii-DFINITY** [comment]
Note that I assume you added the .map for the best performances, that's cool.

Indeed, I could have return the `nonPositiveBalances` directly after checking balance data, but tecnically it is one if condition less (the one that splits into positive and negative balances.

**Review by peterpeterparker — APPROVED** [review]
Sorry, forgot to click "approve" when I reviewed it earlier today. Just added a comment but, it was already fine.

**Pull request merged**